extensions/todo/src/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { ActionPanel, getSelectedFinderItems, Icon, List } from '@raycast/api';
+import { useEffect, useState } from 'react';
+import { create_todo, edit_todo, has_text, Todo } from './todo';
+import { load_todos, store_dones, store_todos } from './persist';
+import { CreateTodoAction, CreateTopPriorityTodoAction } from './todo-create.component';
+import { TodoListItems } from './todo-list-items.component';
+import { to_backup } from './backup';
+import { RestoreBackupAction } from './backup-actions.component';
+
+export default function Command() {
+  // todos
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [dones, setDones] = useState<Todo[]>([]);
+
+  // the app is loading when loading todos
+  const [loading, setLoading] = useState<boolean>(true);
```

Comment:
Note that you can use `useCachedPromise` from `@raycast/utils` for something like that.